src/core/transport/chttp2_transport.c

```diff
@@ -823,24 +823,26 @@ static void unlock(transport *t) {
   finish_reads(t);
 
   /* gather any callbacks that need to be made */
-  if (!t->calling_back && cb) {
+  if (!t->calling_back) {
     perform_callbacks = prepare_callbacks(t);
     if (perform_callbacks) {
```

Comment:
Also, while I'm here, unless prepare_callbacks can change the value of t->calling_back, that `if` here is superfluous, but can't be compiler-optimized.